pkg/controller/servicemesh/memberroll/controller.go

```diff
@@ -151,20 +152,31 @@ func add(mgr manager.Manager, r *MemberRollReconciler) error {
 func toRequests(ctx context.Context, cl client.Client, namespace string) []reconcile.Request {
 	log := common.LogFromContext(ctx)
 	list := &maistrav1.ServiceMeshMemberRollList{}
-	err := cl.List(ctx, list, client.MatchingFields{"spec.members": namespace})
+	err := cl.List(ctx, list)
 	if err != nil {
 		log.Error(err, "Could not list ServiceMeshMemberRolls")
 	}
 
 	var requests []reconcile.Request
 	for _, smmr := range list.Items {
-		requests = append(requests, reconcile.Request{
-			NamespacedName: common.ToNamespacedName(&smmr),
-		})
+		if smmr.Spec.IsClusterScoped() || contains(smmr.Spec.Members, namespace) {
```

Comment:
I don't understand this condition. First condition will return true in cluster-wide mode and the second one will return true in multi-tenant mode, so why we need to check this condition if there is no other alternative?